src/lib/swapper/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -178,3 +183,19 @@ export async function getTradeQuote(
     )
   }
 }
+
+let lifiChainMapPromise: Promise<Result<Map<ChainId, ChainKey>, SwapErrorRight>> | undefined
+
+export const getLifiTradeQuote = async (
+  input: GetEvmTradeQuoteInput,
+  assets: Partial<Record<AssetId, Asset>>,
+  sellAssetPriceUsdPrecision: string,
+): Promise<Result<LifiTradeQuote<false>, SwapErrorRight>> => {
+  if (lifiChainMapPromise === undefined) lifiChainMapPromise = getLifiChainMap()
+
+  const maybeLifiChainMap = await lifiChainMapPromise
+
+  if (maybeLifiChainMap.isErr()) return Err(maybeLifiChainMap.unwrapErr())
+
+  return getTradeQuote(input, maybeLifiChainMap.unwrap(), assets, sellAssetPriceUsdPrecision)
+}
```

Comment:
If this is intended to be a singleton, we might want to move it to it's own file as we've done elsewhere so it's explicit.

Reply:
good call, ill write a todo for this because I want more visibility on the pattern before diving in. Im thinking we have a static class to call these functions with a lazy init of the state data like above